### PR TITLE
New package: VeraCrypt-1.21

### PR DESCRIPTION
**Pull request by GloryS**

### CODE REVIEW

File: srcpkgs/VeraCrypt/template [hunk 1]
```diff
@@ -0,0 +1,40 @@
+# Template file for 'VeraCrypt'
+pkgname=VeraCrypt
+_appname=veracrypt
+version=1.21
+revision=1
+create_wrksrc=yes
+only_for_archs="i686 x86_64"
```
Comment (NeoChontrix) [inline srcpkgs/VeraCrypt/template]: Works fine on Musl and nocross is probably more appropriate.

`nocross="wxWidget"`

File: srcpkgs/VeraCrypt/template [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+build_style=gnu-makefile
+hostmakedepends="wxWidgets wxWidgets-devel yasm fuse fuse-devel pkg-config makeself"
+makedepends="wxWidgets wxWidgets-devel yasm fuse fuse-devel pkg-config makeself"
+depends="wxWidgets fuse"
```
Comment (NeoChontrix) [inline srcpkgs/VeraCrypt/template]: `hostmakedepends="yasm pkg-config"
makedepends="wxWidgets-devel fuse-devel"`
This should be enough.

Comment (GloryS) [inline srcpkgs/VeraCrypt/template]: You right, this enough.

File: srcpkgs/VeraCrypt/template (hunk 1 above)
Comment (GloryS) [inline srcpkgs/VeraCrypt/template]: Hmm...What about arm? Developer mentions only x86.
https://www.veracrypt.fr/en/Supported%20Operating%20Systems.html

Comment (NeoChontrix) [inline srcpkgs/VeraCrypt/template]: OK, haven't checked the site. Then you may only have to add x86-64-musl.

Comment (GloryS) [inline srcpkgs/VeraCrypt/template]: Add musl archs...

File: srcpkgs/VeraCrypt/template [hunk 3]
```diff
@@ -0,0 +1,40 @@
+# Template file for 'VeraCrypt'
+pkgname=VeraCrypt
+_appname=veracrypt
+version=1.21
+revision=1
+create_wrksrc=yes
+only_for_archs="i686 x86_64 i686-musl x86_64-musl"
```
Comment (the-maldridge) [inline srcpkgs/VeraCrypt/template]: This is almost certainly wrong.  Fix the cross builds or if it truly can't be cross compiled mark it broken for the applicable architectures.

File: srcpkgs/VeraCrypt/template [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+depends="wxWidgets fuse"
+short_desc="VeraCrypt is a free open source disk encryption software"
+maintainer="Viatcheslav E. Kouznetsov <glorysmith@gmail.com>"
+license="Apache 2.0"
+homepage="https://www.veracrypt.fr/en"
+_filename="${pkgname}_${version}_Source.tar.bz2"
+distfiles="https://launchpadlibrarian.net/328011925/${_filename}"
+checksum=44717d24b8deff2db9dd0365572063e38bbe4a59438a8c3c07675b5eed370358
+
+do_extract() {
```
Comment (the-maldridge) [inline srcpkgs/VeraCrypt/template]: This looks fishy, you should probably be setting wrksrc to what this extracts to instead.

File: srcpkgs/VeraCrypt/template [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+	tar xf ${XBPS_SRCDISTDIR}/${pkgname}-${version}/${_filename} \
+	  -C ${wrksrc}/
+}
+
+do_build() {
```
Comment (the-maldridge) [inline srcpkgs/VeraCrypt/template]: No, you should set build_wrksrc instead.  This breaks all the flags that should be passed here.

File: srcpkgs/VeraCrypt/template [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+	cd ${wrksrc}/src
+	make
+}
+
+do_install() {
+	cd ${wrksrc}/src
+	vbin Main/${_appname}
+	vbin Setup/Linux/${_appname}-uninstall.sh
+	vdoc License.txt
```
Comment (the-maldridge) [inline srcpkgs/VeraCrypt/template]: No, this should be vlicense not vdoc so that it winds up in the right place.

**Comment by maxice8** [comment]
@the-maldridge VeraCrypt has already been updated to 1.22

**Pull request closed without merging**